improver/threshold.py

```diff
@@ -248,7 +248,7 @@ def _add_threshold_coord(self, cube: Cube, threshold: float) -> None:
                 Value at which the data has been thresholded
         """
         coord = iris.coords.DimCoord(
-            np.array([threshold], dtype=FLOAT_DTYPE), units=cube.units
+            np.array([threshold], dtype="float64"), units=cube.units
```

Comment:
Similarly, it might be worth adding a comment or updating the docstring to include something like `Ensure the threshold is float64 to avoid rounding errors during possible unit conversion.`